RELEASE.md

```diff
@@ -0,0 +1,55 @@
+# OpenTelemetry Release Process
+
+This repository uses semantic versioning. Please keep this in mind when choosing version numbers.
+
+1. **Alert others you are releasing**
+
+   There should be no commits made to master while the release is in progress (about 10 minutes). Before you start
+   a release, alert others on [gitter](https://gitter.im/open-telemetry/opentelemetry-java) so that they don't accidentally
+   merge anything. If they do, and the build fails because of that, you'll have to recreate the release tag described below.
+
+1. **Update CHANGELOG**
+
+   Update CHANGELOG.md with a list changes since the last release. Each entry must include the release number,
+   date and a bulleted list of changes where each change is summarized in a single sentence.
+
+1. **Update the version**
+
+   Update the version to the desired value in `build.gradle`, e.g. `version = "1.2.3"`. Make sure no `SNAPSHOT`
+   sufix is appended (that is **only** used during development and when deploying snapshots, as the word implies).
+   Commit, merge to `master`, and update your local `master` branch with this update.
+
+1. **Push a git tag**
+
+   The tag should be of the format `vN.M.L`, e.g. `git tag v1.2.3; git push origin v1.2.3`.
+
+1. **Wait for Circle CI**
```

Comment:
Why not use a release branch?

Reply:
Actually there's no restriction on doing it from a branch - also, when we have the full automatized version part, it won't even matter. Does that resolve your comment?

Reply:
Let's recommend that. I prefer that way because it removes the race condition of somebody else merging during the same time (a bit earlier).